base/util_testing.go

```diff
@@ -538,20 +535,34 @@ func NumOpenBuckets(bucketName string) int32 {
 	return numOpen
 }
 
-// EnableTestLogKey will enable the given log key, if it exists,
-// and also enable debug level logging, if a + log key is provided.
-func EnableTestLogKey(logKey string) {
-	if strings.HasSuffix(logKey, "+") {
-		ConsoleLogLevel().Set(LevelDebug)
+// SetUpTestLogging will set the given log level and log keys,
+// and return a function that can be deferred for teardown.
+//
+// To set multiple log keys, use the bitwise OR operator.
+// E.g. KeyCache|KeyDCP|KeySync
+//
+// Usage:
+//     teardownFn := SetUpTestLogging(LevelDebug, KeyCache|KeyDCP|KeySync)
+//     defer teardownFn()
+//
+// Shorthand style:
+//     defer SetUpTestLogging(LevelDebug, KeyCache|KeyDCP|KeySync)()
+func SetUpTestLogging(logLevel LogLevel, logKeys LogKey) (teardownFn func()) {
+	Infof(KeyAll, "Test: Set log level: %v", logLevel)
+	Infof(KeyAll, "Test: Set log keys: %v", logKeys)
+
+	// Set the console logger settings
+	consoleLogger.LogLevel.Set(logLevel)
+	consoleLogger.LogKey.Set(logKeys)
+
+	return func() {
+		// Return to default state
+		consoleLogger.LogLevel.Set(LevelInfo)
+		consoleLogger.LogKey.Set(KeyHTTP)
+
+		Infof(KeyAll, "Test: Reset log level: %v", LevelInfo)
+		Infof(KeyAll, "Test: Reset log keys: %v", KeyHTTP)
```

Comment:
As discussed, the callback logging may be useful to identify tests that fail to invoke the callback.  To make that more useful, would want to be able to log the calling test name (or some other identifier that at least lets us try to pair up 'setup' and 'teardown' logging.